# HG changeset patch
# User Rik <rik@octave.org>
# Date 1478186650 25200
#      Thu Nov 03 08:24:10 2016 -0700
# Branch stable
# Node ID e07fa946f7631c89cda700bbaff9f3197b1ca247
# Parent  413a19bca7f37082e042a5add7e13a9afee17163
Fix regression in feval and emit an error when function does not exist (bug #49516).

* oct-parse.in.yy (feval): If fcn is not defined, call error() rather
than trying maybe_missing_function_hook which does not produce warning
or return a status code if function does not exist.

diff --git a/libinterp/parse-tree/oct-parse.in.yy b/libinterp/parse-tree/oct-parse.in.yy
--- a/libinterp/parse-tree/oct-parse.in.yy
+++ b/libinterp/parse-tree/oct-parse.in.yy
@@ -4781,26 +4781,17 @@ feval (const std::string& name, const oc
 {
   octave_value_list retval;
 
   octave_value fcn = symbol_table::find_function (name, args);
 
   if (fcn.is_defined ())
     retval = fcn.do_multi_index_op (nargout, args);
   else
-    {
-      try
-        {
-          maybe_missing_function_hook (name);
-        }
-      catch (octave::execution_exception& e)
-        {
-          error (e, "feval: function '%s' not found", name.c_str ());
-        }
-    }
+    error ("feval: function '%s' not found", name.c_str ());
 
   return retval;
 }
 
 octave_value_list
 feval (octave_function *fcn, const octave_value_list& args, int nargout)
 {
   octave_value_list retval;
